### ISSUE TITLE
Cop::AutocorrectAlignment#calculate_range returns non sensible range.

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#5084]
Pretty minor issue that sadly I don't have time to investigate further or fix right now but I thought I'd mention it.

Currently, neither `rubocop` nor `parser` check that a range being rewritten is within the bounds of the source.

If you add the check (like [in this commit](https://github.com/marcandre/rubocop/commit/cfbf84a1522bb1639a232a4ca2c960a38bbbccba)), you'll get warnings with 9 specs, all of which seem to be due to this `Cop::AutocorrectAlignment#calculate_range` returning a range that ends past the source. So clearly the range is non-sensical, at least in some circumstances.

For the rewriting itself, it's not a big issue since it's used with `insert_before`, but the range is also used to check with `heredoc_ranges`, so there could be false positives there.

The specs triggering the warnings:
```
rspec ./spec/rubocop/cli/cli_autocorrect_spec.rb:643 ./spec/rubocop/cop/layout/indent_array_spec.rb:405 ./spec/rubocop/cop/layout/indent_array_spec.rb:238 ./spec/rubocop/cli/cli_options_spec.rb:171 ./spec/rubocop/cop/layout/indent_hash_spec.rb:313 ./spec/rubocop/cop/layout/indent_hash_spec.rb:472 ./spec/rubocop/cop/layout/align_parameters_spec.rb:348 ./spec/rubocop/cop/layout/align_parameters_spec.rb:312 ./spec/rubocop/cop/layout/closing_parenthesis_indentation_spec.rb:235 ./spec/rubocop/cop/layout/closing_parenthesis_indentation_spec.rb:55 ./spec/rubocop/cop/layout/closing_parenthesis_indentation_spec.rb:163 
```

**Comment by Drenmi** [comment]
Thanks for the investigation @marcandre! 🙂

**Comment by marcandre** [comment]
See aed2c13ee29958268a and b4dd314125345b7c40
